src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -167,6 +166,22 @@ def setUp(self):
         self.log.info("*** SETUP running on %s ***", str(detect()))
         super().setUp()
 
+        # Random generator that could be seeded for reproducibility
+        env_seed = os.environ.get("DAOS_TEST_RANDOM_SEED", None)
+        if env_seed is None:
+            self.rand_seed = int.from_bytes(os.urandom(8), byteorder='little')
+        else:
+            try:
+                self.rand_seed = int(env_seed)
+            except ValueError:
+                self.log.error(
+                    "ERROR: The env variable DAOS_TEST_RANDOM_SEED "
+                    "does not define a valid integrer: got='%s'",
+                    env_seed)
+                raise
```

Comment:
Minor - in case this fails, reporting is better if we use self.fail
```suggestion
            except ValueError:
                self.fail(
                    "The env variable DAOS_TEST_RANDOM_SEED "
                    "does not define a valid integrer: got='{}'".format(env_seed))
```

Reply:
I was originally using `self.fail()` but it was generating some two nested stack traces and thus it was looking more friendly like this.
However, respecting ftest coding guidelines is more important and thus I will update it as you recommended.

- [ ] s/raise/fail/

Reply:
Fixed with commit https://github.com/daos-stack/daos/pull/13013/commits/270805402e811b5f153193363cf4304b7dcc26f0#diff-aa5a03545fb0ad9ec37f79bb9c06e107ded1fee8997c345356dcbd6d51c42fcaL55